### PR TITLE
refactor: change freq_norm from string to enum

### PR DESCRIPTION
**Pull request by anujsinha3**
https://github.com/uw-ssec/codeuw/issues/20

### CODE REVIEW

File: script/test_whiten.py [hunk 1]
```diff
@@ -3,7 +3,8 @@
 import scipy
 from scipy.fftpack import next_fast_len
 
-from noisepy.seis.noise_module import moving_ave, whiten
+from src.noisepy.seis.datatypes import FreqNorm
+from src.noisepy.seis.noise_module import moving_ave, whiten
```
Comment (carlosgjs) [inline script/test_whiten.py:7]: hmm, the `src.` prefix shouldn't be required. Did you `pip install -e .` as described in the CONTRIBUTING.md file?

File: src/noisepy/seis/datatypes.py [hunk 2]
```diff
@@ -139,7 +145,7 @@ class ConfigParameters(BaseModel):
     freqmin: float = Field(default=0.05)
     freqmax: float = Field(default=2.0)
     freq_norm: str = Field(
```
Comment (carlosgjs) [inline src/noisepy/seis/datatypes.py:147]:
```suggestion
    freq_norm: FreqNorm = Field(
```

**Review by carlosgjs — CHANGES_REQUESTED** [review]
Looking great, couple of small corrections. Also, the failure in the notebook is a bit cumbersome to debug, but here's how to find the log:

image

In this case, the error is:
```
[0;31mValidationError[0m: 1 validation error for ConfigParameters
freq_norm
  Input should be a valid string [type=string_type, input_value=<FreqNorm.RMA: 'rma'>, input_type=FreqNorm]
    For further information visit https://errors.pydantic.dev/2.3/v/string_type
```
The `ConfigParameter.freq_form` field was still declared as `str`, needs to be updated to `FreqNorm`. 

Also note that you can open the notebook in VS Code and run it directly there and the error should show up.

File: tests/test_whiten.py [hunk 3]
```diff
@@ -158,12 +166,14 @@ def plot_2d(white_original, white_new):
 
 
 # Use wrappers since test functions are not supposed to return values
-def test_whiten1d():
-    _, _ = whiten1d()
+@pytest.mark.parametrize("freqNorm", [FreqNorm.PHASE_ONLY, FreqNorm.RMA])
```
Comment (carlosgjs) [inline tests/test_whiten.py:169]: very nice! And I see that the code coverage check is now passing.

**Comment by anujsinha3** [comment]
@carlosgjs: I am unable to find the logs for the notebook failure. 

I am getting the below warning-error logs from the build pages log:
https://github.com/noisepy/NoisePy/actions/runs/6647272604/job/18062357031?pr=248#step:4:285


Also, I am not able to find the artifact log files as well. The picture you posted looks a bit distorted. Can you please clarify once again?

File: script/test_whiten.py (hunk 1 above)
Comment (anujsinha3) [reply script/test_whiten.py]: Done.

File: src/noisepy/seis/datatypes.py (hunk 2 above)
Comment (anujsinha3) [reply src/noisepy/seis/datatypes.py]: Done

**Review by carlosgjs — APPROVED** [review]
Looks good, thanks!

**Pull request merged**